### PR TITLE
refine the quantization doc

### PR DESCRIPTION
**Pull request by yufenglee**

### CODE REVIEW

**Review by zhanghuanrong — APPROVED** [review]

File: docs/how-to/quantization.md [hunk 1]
```diff
@@ -41,76 +41,26 @@ There are 3 ways of quantizing a model: dynamic, static and quantize-aware train
...
 
-* **Optimizing a quantized model**
+If both post-training quantization can not meet your accuracy goal, you can try quantization-aware training to retrain the model. OnnxRuntime doesn't provide retrain capability now. You can retrain you model with original framework and then converted it back to ONNX. 
 
-    The same holds the other way around. After quantizing a model, some graph optimizations which otherwise might have been applicable may not be applicable anymore.
+## List of Supported Quantized Ops
+Please refer to [registry](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/quantization/registry.py) for the list of supported Ops.
 
+## Quantization and model opset versions
+ Quantization ops were introduced in ONNX opset version 10, so the model which is being quantized must be opset 10 or higher. If the model opset version is < 10 then the model should be reconverted to ONNX from its original framework using a later opset.
```
Comment (faxu) [inline docs/how-to/quantization.md:55]: We can keep it short and just say that opset 10 is required. 
"Models must be opset10 or higher to be quantized. Models with opset < 10 must be reconverted to ONNX from its original framework using a later opset."

File: docs/how-to/quantization.md [hunk 2]
```diff
@@ -41,76 +41,26 @@ There are 3 ways of quantizing a model: dynamic, static and quantize-aware train
...
 
 ## Quantization API
-Quantization has 3 main APIs:
+Quantization has 3 main APIs, which corresponds to the 3 quantization method:
```
Comment (faxu) [inline docs/how-to/quantization.md:58]: method**s**

File: docs/how-to/quantization.md [hunk 3]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+### ARM64
+U8S8 can be faster than U8U8 for low end ARM64 and no difference on accuracy. There is no difference for high end ARM64.
+
+## Transformer-based models
+We have specific optimization for transformer-based models, like QAttention for quantization of attention layer. In order to leverage those specific optimization, you need to optimize your models with [Transformer Model Optimization Tool](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers) before quantizing the model.
+
+We have a [notebook](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/quantization/notebooks/bert) that demonstrates the E2E process.
+
+## Quantization on GPU
+Hardware suppor is required to achieve better performance with quantization on GPUs. You need a device that supports Tensor Core int8 computation, like T4, A100. Older hardware won't get benefit.
```
Comment (faxu) [inline docs/how-to/quantization.md:124]: suppor**t**

File: docs/how-to/quantization.md [hunk 4]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+
+We leverage TRT EP for quantization on GPU. Different with CPU EP, TRT takes in full precision model and calibration result for inputs. It decides how to quantize with their own logic. The overall procedure to leverage TRT EP quantization is:
```
Comment (faxu) [inline docs/how-to/quantization.md:126]: let's link TRT EP to relative link: ../../reference/execution-providers/TensorRT-ExecutionProvider.md

File: docs/how-to/quantization.md [hunk 5]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+- Implement a [CalibrationDataReader](https://github.com/microsoft/onnxruntime/blob/07788e082ef2c78c3f4e72f49e7e7c3db6f09cb0/onnxruntime/python/tools/quantization/calibrate.py).
+- Compute quantization parameter with calibration data set. Our quantization tool supports 2 calibration methods: MinMax and Entropy. Note: In order to include all tensors from the model for better calibration, please run symbolic_shape_infer.py first. Please refer to[here](https://www.onnxruntime.ai/docs/reference/execution-providers/TensorRT-ExecutionProvider.html#sample) for detail.
+- Save quantization parameter into a flatbuffer file
+- Load model and quantization parameter file and run with TRT EP.
+
+We have 2 E2E examples [Yolo V3](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/quantization/E2E_example_model/object_detection/trt/yolov3) and [resnet50](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/quantization/E2E_example_model/image_classification/trt/resnet50) for your reference.
+
+## FAQ
+### Why performance is not better or even worse?
```
Comment (faxu) [inline docs/how-to/quantization.md:135]: Suggestion: "Why am I not seeing performance improvements?"

File: docs/how-to/quantization.md [hunk 6]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+## Which quantization data type to use?
+Quantization represents value with 8 bit, which can be either int8 and uint8. Combining with activation and weight, the data format can be (activation:uint8, weight:uint8), (activation:uint8, weight:int8), etc.
+
+Let's use U8U8 as as shorthand for (activation:uint8, weight:uint8), and U8S8 for (activation:uint8, weight:int8), and S8U8, S8S8 for other two formats.
+
+Currently, OnnxRuntime CPU only supports activation with type uint8, i.e., U8X8 only.
+
+### x86-64
+#### AVX2
+Try U8U8 firstly, and then U8S8.
```
Comment (faxu) [inline docs/how-to/quantization.md:99]: "first"

File: docs/how-to/quantization.md [hunk 7]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+- Performance
+U8S8 leverage VPMADDUBSW instruction but U8U8 kernel can process 6 rows at a time versus 4 rows for the U8S8 kernel, and U8U8 sequence is two instructions vs three instructions for U8S8. Thus, this balance ends up with u8u8 just in reach of U8S8 for older HW (Broadwell)
+- Accuracy
+VPMADDUBSW has saturation issue, thus U8S8 needs to use [reduce_range](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/quantization/quantize.py) if accuracy is not good enough.
+
+#### AVX512
+U8S8 is much faster, but may suffer saturation issue.
+- Performance
+U8S8 can only do twice mul/adds per instruction as much as U8U8. Typically U8S8 sequence ends up winning, can be twice as fast as u8u8.
```
Comment (faxu) [inline docs/how-to/quantization.md:108]: "U8S8 can do twice as many mul/adds per instruction compared to U8U8 so can be twice as fast."

File: docs/how-to/quantization.md [hunk 8]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+- Accuracy
+Same as the AVX2 because of VPMADDUBSW. Needs to ure reduce_range if accuracy is not good enough.
```
Comment (faxu) [inline docs/how-to/quantization.md:110]: "use" typo

File: docs/how-to/quantization.md [hunk 9]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+
+#### VNNI
+No difference.
+
+### ARM64
+U8S8 can be faster than U8U8 for low end ARM64 and no difference on accuracy. There is no difference for high end ARM64.
+
+## Transformer-based models
+We have specific optimization for transformer-based models, like QAttention for quantization of attention layer. In order to leverage those specific optimization, you need to optimize your models with [Transformer Model Optimization Tool](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers) before quantizing the model.
```
Comment (faxu) [inline docs/how-to/quantization.md:119]: Let's change "we have" to "there are" and in general use more objective wording. There are other areas that use "we" that can also be reworded"

File: docs/how-to/quantization.md [hunk 10]
```diff
@@ -137,31 +87,60 @@ quantized_model = quantize_qat(model_fp32, model_quant)
...
+
+We have a [notebook](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/quantization/notebooks/bert) that demonstrates the E2E process.
```
Comment (faxu) [inline docs/how-to/quantization.md:121]: "This notebook demonstrates..."

**Review by faxu — APPROVED** [review]

**Review by faxu — APPROVED** [review]

**Pull request merged**